src/tools/pkcs11-tool.c

```diff
@@ -5203,6 +5205,10 @@ show_key(CK_SESSION_HANDLE sess, CK_OBJECT_HANDLE obj)
 		printf("%ssign", sepa);
 		sepa = ", ";
 	}
+	if ((!pub || sec) && getSIGN_RECOVER(sess, obj)) {
```

Comment:
the `SIGN_RECOVER` is not defined for secret keys (unlike the SIGN, which is):
```suggestion
	if (!pub && getSIGN_RECOVER(sess, obj)) {
```

Reply:
OK, actually its defined for couple of  `CKM_SEED_*` mechanisms but I hope nobody implements these so I would propose to not print this for generic secret keys.

http://docs.oasis-open.org/pkcs11/pkcs11-curr/v2.40/cs01/pkcs11-curr-v2.40-cs01.html#_Toc399399104

Reply:
Thank you for the clarification! Change committed.